modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -2456,6 +2457,48 @@ public void visit(LambdaExpression lambdaExpr) {
     }
 
     @Override
+    public void visit(StringTemplateLiteral stringTemplateLiteral) {
+        Expression[] items = stringTemplateLiteral.getItems();
+        List<Expression> newItems = new ArrayList<>();
+        // Consecutive non-string type items are converted to string, and combined together using binary add
+        // expressions.
+        Expression addExpr = null;
```

Comment:
Create the complete BinaryAddExpression first and then evaluate the complete expression.